src/lib/rmt/wizard_final_page.rb

```diff
@@ -86,4 +87,19 @@ def run
     render_content
     event_loop
   end
+
+  private
+
+  def firewall_status
+    firewalld = Y2Firewall::Firewalld.instance
+    services = ['http', 'https']
+
+    return _('(firewalld not enabled)') unless firewalld.enabled?
+
+    if firewalld.find_zone(firewalld.default_zone).services & services == services
```

Comment:
Not sure about this, I mean, what if the active interfaces are not in the default zone but in other one which also have the services opened.

And also the checking is more safety if called as `services & public_services == services` because the list of allowed services is not sorted if I'm not wrong.
```ruby
[27] pry(main)> ["http", "https"] & ["https", "ftp", "http"] == ["http", "https"]
=> true
[28] pry(main)> ["https", "ftp", "http"] & ["http", "https"] == ["http", "https"]
=> false
```

Reply:
> Not sure about this, I mean, what if the active interfaces are not in the default zone but in other one which also have the services opened.

Hm, what would then be the "catch them all" solution?

> And also the checking is more safety if called as `services & public_services == services` because the list of allowed services is not sorted if I'm not wrong.

Oh sounds you just saved us from a hard to find bug of the future :robot:

Reply:
Mmm probably it should be better something like

```ruby
fw.zones.any? { |z| (z.services & services).size == services.size }
```

And even in this case I'm not sure. Maybe would make sense to make `public` the CWMFirewallInterfaces.current_firewall_status method

https://github.com/yast/yast-yast2/blob/master/library/network/src/modules/CWMFirewallInterfaces.rb#L960

Reply:
@teclator 
The `current_firewall_status` doesn't seem to be the right approach to me :thinking: Since it would only return `open_all` if that service was allowed on all interfaces. So if a user would use the "Details" of our firewall widget and only select one of multiple network cards (interfaces), the  `current_firewall_status` method wouldn't return "open", or would it?

Iterating over all zones and showing "Open" if it is open in some zone could be considered "good enough" I think, since RMT is usually installed not alongside with many other web services on a single host, but mostly as the only service on a machine. At least our nginx setup asumes that there is nothing else using nginx (afair).

I just tried that, replacing line 99 with this:
```ruby
if firewalld.zones.any? { |zone| services & zone.services == services }
```
and it behaves as expected on my test machine.

But now I'm wondering about the internals of `OpenFirewallWidget`... Which zone does it actually change? The default zone I assume? If so, the initial version (as-is now) would be more correct, or wouldn't it?

Reply:
> @teclator
> The `current_firewall_status` doesn't seem to be the right approach to me thinking Since it would only return `open_all` if that service was allowed on all interfaces. So if a user would use the "Details" of our firewall widget and only select one of multiple network cards (interfaces), the `current_firewall_status` method wouldn't return "open", or would it?

It will return `:custom`
https://github.com/yast/yast-yast2/blob/master/library/network/src/modules/CWMFirewallInterfaces.rb#L968

which, when transforming to a label means `open only in the selected interfaces`

https://github.com/yast/yast-yast2/blob/master/library/network/src/modules/CWMFirewallInterfaces.rb#L999

> 
> Iterating over all zones and showing "Open" if it is open in some zone could be considered "good enough" I think, since RMT is usually installed not alongside with many other web services on a single host, but mostly as the only service on a machine. At least our nginx setup asumes that there is nothing else using nginx (afair).
> 
> I just tried that, replacing line 99 with this:
> 
> ```ruby
> if firewalld.zones.any? { |zone| services & zone.services == services }
> ```
> 
> and it behaves as expected on my test machine.
> 
> But now I'm wondering about the internals of `OpenFirewallWidget`... Which zone does it actually change? The default zone I assume? If so, the initial version (as-is now) would be more correct, or wouldn't it?

In firewalld you usually allow services in specific zones and then assign the interfaces to that zones. If no zone is specified explicitly then the default one is used.

In case of CWMFIrewallInterfaces, you choose in which interfaces the services should be allowed which is later translated to zones, so it depends in the current configuration being the `default zone` in case of not previous changes.

Probably in the future we will provide a new CWM widget which allows to open the services only in the active zones or things like that, but for the time being this is what we can offer by now. I hope I clarified it instead of making it more complicate to understand.

Reply:
@teclator I just pushed aa1f4b96b1ca96301515cf7f2d405c5eb1ce7eaa to use the widget's label logic. Wanna have look before I call it done? :wink: